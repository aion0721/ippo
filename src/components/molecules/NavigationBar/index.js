import React from 'react';
import { Navbar } from '@blueprintjs/core';
import PropTypes from 'prop-types';

const NavigationBar = ({ appName }) => (
  <Navbar>
    <Navbar.Group>
      <Navbar.Heading>{appName}</Navbar.Heading>
    </Navbar.Group>
  </Navbar>
);

NavigationBar.displayName = 'NavigationBar';

NavigationBar.propTypes = {
  appName: PropTypes.string.isRequired,
};

export default NavigationBar;
