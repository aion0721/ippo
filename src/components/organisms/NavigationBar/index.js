import React from 'react';
import { Alignment, Button, Navbar } from '@blueprintjs/core';
import propTypes from 'prop-types';

const NavigationBar = ({ history }) => (
  <Navbar>
    <Navbar.Group align={Alignment.LEFT}>
      <Navbar.Heading>IPPO</Navbar.Heading>
      <Navbar.Divider />
    </Navbar.Group>
    <Navbar.Group align={Alignment.RIGHT}>
      <Button icon="wrench" minimal onClick={() => history.push('/notifications/publish')} />
    </Navbar.Group>
  </Navbar>
);

NavigationBar.displayName = 'NavigationBar';

NavigationBar.propTypes = {
  history: propTypes.object.isRequired,
};

export default NavigationBar;
