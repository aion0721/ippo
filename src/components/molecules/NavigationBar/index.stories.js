import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import NavigationBar from '.';

const stories = storiesOf('molecules/NavigationBar', module);

stories.add('通常パターン', () => <NavigationBar appName={text('appName', 'タイトル')} />);
