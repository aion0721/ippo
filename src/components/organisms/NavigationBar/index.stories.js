import React from 'react';
import { storiesOf } from '@storybook/react';
import NavigationBar from '.';

const stories = storiesOf('organisms/NavigationBar', module);

stories.add('通常パターン', () => <NavigationBar history={{}} />);
