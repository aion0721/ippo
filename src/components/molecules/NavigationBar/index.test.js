import React from 'react';
import { snapshot } from 'test/helpers';
import NavigationBar from '.';

const props = {
  appName: 'test',
};

snapshot('NavigationBar/nomal', <NavigationBar {...props} />);
