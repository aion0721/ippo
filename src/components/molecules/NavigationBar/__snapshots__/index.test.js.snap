// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NavigationBar/nomal 1`] = `
<NavigationBar
  appName="test"
>
  <Blueprint3.Navbar>
    <div
      className="bp3-navbar"
    >
      <Blueprint3.NavbarGroup
        align="left"
      >
        <div
          className="bp3-navbar-group bp3-align-left"
        >
          <Blueprint3.NavbarHeading>
            <div
              className="bp3-navbar-heading"
            >
              test
            </div>
          </Blueprint3.NavbarHeading>
        </div>
      </Blueprint3.NavbarGroup>
    </div>
  </Blueprint3.Navbar>
</NavigationBar>
`;
